ext/phar/phar_object.c

```diff
@@ -5370,8 +5405,8 @@ void phar_object_init(TSRMLS_D) /* {{{ */
 	phar_ce_archive = zend_register_internal_class(&ce TSRMLS_CC);
 	phar_ce_archive->ce_flags |= ZEND_ACC_FINAL_CLASS;
```

Comment:
I'm not sure it's a good practice to extend a final class. In general, if we extend it maybe it should not be final, but if we need it to be final, maybe set it to final after we extend it.